pydrive2/files.py

```diff
@@ -97,6 +98,48 @@ def _GetList(self):
         return result
 
 
+class FakeWriteable(object):
+    def write(self, chunk):
+        self.chunk = chunk
+
+
+class MediaIoReadable(object):
+    def __init__(self, request, chunksize=DEFAULT_CHUNK_SIZE, pre_buffer=True):
+        """File-like wrapper around MediaIoBaseDownload.
+
+    :param pre_buffer: Whether to read one chunk into an internal buffer
+    immediately in order to raise any potential errors.
+    :raises: HttpError
+    """
+        self.done = False
+        self.fd = FakeWriteable()
+        self.downloader = MediaIoBaseDownload(
+            self.fd, request, chunksize=chunksize
+        )
+        self._buffer = None
+        if pre_buffer:
+            self._buffer = self.read()
+
+    def read(self, chunksize=DEFAULT_CHUNK_SIZE):
```

Comment:
could also do:

```suggestion
    def read(self, chunksize=None):
```

Reply:
Not sure I got the comment ...

Reply:
having `chunksize=DEFAULT_CHUNK_SIZE` means by default every single call we will set a private member `downloader._chunksize`. This private member may at some point cease to be used if the upstream library changes. Instead using `chunksize=None` will leave everything to its defaults... by default.

Reply:
it feels I'm missing something still, sorry ... why do we need to provide `chunksize=None` at all? how setting default to None changes the fact that access private member

Reply:
I assume users would like `chunksize` as an optional argument to `read()`. It's best to have as an argument to the constructor (as is) and not constantly reset it in `read()` but perhaps people would like the option of controlling variable `chunksize`?

By setting default `None`, we don't try to constantly access the private member (unless the user specifies some value manually).

Reply:
k, gotcha ... for me was hard to grasp the value of "messing with private member by _default_" I suppose (i.e. it's better to not access it at all, right? :))

to be honest I would just get rid of this for now :) ... feels a bit premature and since we provide ctr param It should be more than enough as a first step.

Reply:
sure I don't mind.